lib/Swagger/src/User/Preferences/Api/UserPreferencesApi.php

```diff
@@ -0,0 +1,342 @@
+      $apiClient = new ApiClient();
+      $apiClient->getConfig()->setHost('https://localhost/user-preference');
+    }
+
+    $this->apiClient = $apiClient;
+  }
+
+  /**
+   * @return \Swagger\Client\ApiClient get the API client
+   */
+  public function getApiClient() {
+    return $this->apiClient;
+  }
+
+  /**
+   * @param \Swagger\Client\ApiClient $apiClient set the API client
+   * @return UserPreferencesApi
+   */
+  public function setApiClient(ApiClient $apiClient) {
+    $this->apiClient = $apiClient;
+    return $this;
+  }
+
+  
+  /**
+   * getUserPreferences
+   *
+   * Returns all the global user preferences for a user
+   *
+   * @param int $user_id The id of the user to list the preferences (required)
+   * @return \Swagger\Client\User\Preferences\Models\Preference[]
+   * @throws \Swagger\Client\ApiException on non-2xx response
+   */
+   public function getUserPreferences($user_id) {
+      
+      // verify the required parameter 'user_id' is set
+      if ($user_id === null) {
+        throw new \InvalidArgumentException('Missing the required parameter $user_id when calling getUserPreferences');
+      }
+      
+
+      // parse inputs
+      $resourcePath = "/{userId}";
+      $resourcePath = str_replace("{format}", "json", $resourcePath);
+      $method = "GET";
+      $httpBody = '';
+      $queryParams = array();
+      $headerParams = array();
+      $formParams = array();
+      $_header_accept = ApiClient::selectHeaderAccept(array('application/json'));
+      if (!is_null($_header_accept)) {
+        $headerParams['Accept'] = $_header_accept;
+      }
+      $headerParams['Content-Type'] = ApiClient::selectHeaderContentType(array());
+
+      
+      
+      // path params
+      if($user_id !== null) {
+        $resourcePath = str_replace("{" . "userId" . "}",
+                                    $this->apiClient->getSerializer()->toPathValue($user_id),
+                                    $resourcePath);
+      }
+      
+      
+
+      // for model (json/xml)
+      if (isset($_tempBody)) {
+        $httpBody = $_tempBody; // $_tempBody is the method argument, if present
+      } else if (count($formParams) > 0) {
+        // for HTTP post (form)
+        $httpBody = $formParams;
+      }
+      
+      $apiKey = $this->apiClient->getApiKeyWithPrefix('X-Wikia-AccessToken');
+      if (isset($apiKey)) {
+        $headerParams['X-Wikia-AccessToken'] = $apiKey;
+      }
+      
+      
+      
+      $apiKey = $this->apiClient->getApiKeyWithPrefix('X-Wikia-UserId');
+      if (isset($apiKey)) {
+        $headerParams['X-Wikia-UserId'] = $apiKey;
+      }
+      
+      
+      
+      // make the API Call
+      try {
+        $response = $this->apiClient->callAPI($resourcePath, $method,
+                                              $queryParams, $httpBody,
+                                              $headerParams);
+      } catch (ApiException $e) {
+        switch ($e->getCode()) { 
```

Comment:
We can add that in the Swagger documentation and regenerate, but I think that can be a separate PR since we don't do anything with the status code in the exceptions thrown